server/src/modules/boards/boards.service.ts

```diff
@@ -174,22 +180,27 @@ export class BoardsService {
   }
 
   async findUserRelatedBoards(
-    query: FilterQuery<SuperBoard>,
+    query: FilterQuery<SuperBoardDocument>,
     page: number,
     limit: number,
-    order: string,
-  ): Promise<SuperBoardWithoutSlides[]> {
+    order: SortOrder,
+  ): Promise<BoardWithoutSlides[]> {
     const skip = (page - 1) * limit;
+    const sortOrder = order === 'ascending' ? 'ascending' : 'descending';
     return this.boardModel
       .find(query)
-      .sort([['updatedAt', order === 'ascending' ? 'ascending' : 'descending']])
+      .sort([['updatedAt', sortOrder]])
       .skip(skip)
       .limit(limit)
       .select('-slides')
       .exec();
   }
 
-  async queryCountDocuments(query: FilterQuery<SuperBoard>): Promise<number> {
+  async queryCountDocuments(
+    query: FilterQuery<SuperBoardDocument>,
```

Comment:
I didnt know about this approach, why not mentioned before implementing the endpoint?

Reply:
There were no info details about adding permission to filter option, how would i know?

Reply:
i didn't go through your code before, i thought that maybe you have in mind we would want to extend some things in future

Reply:
I was told to implement two approaches 
1) fetch boards for which I am an owner
2) fetch boards that are shared for me

I can extend the implementation but in a different ticket. We can discuss the details on the sync in order to avoid similar events.

Reply:
> There wasno info details about adding permission to filter option, how would i know?

1. That is why it is not needed to be changed in this PR - it is just suggestion to be considered in future
2. I am not saying that specifically permission should have this filter option, but in general those columns